cabal-install/Distribution/Client/Sandbox/PackageEnvironment.hs

```diff
@@ -227,14 +241,45 @@ sandboxPackageDBPath sandboxDir compiler platform =
 -- into a sandbox.
 
 -- | Use the package DB location specific for this compiler.
-setPackageDB :: FilePath -> Compiler -> Platform -> ConfigFlags -> ConfigFlags
-setPackageDB sandboxDir compiler platform configFlags =
-  configFlags {
-    configPackageDBs = [Just (SpecificPackageDB $ sandboxPackageDBPath
+setPackageDB :: FilePath -> Compiler -> Platform -> ConfigFlags ->
+                ProgramConfiguration -> ConfigFlags
+setPackageDB sandboxDir compiler platform configFlags conf =
+  if viewSupported
+    then configFlags {
+      configPackageDBs = [Just UserPackageDB],
+      configView = Flag $ sandboxDirHash sandboxDir
+    }
+    else configFlags {
+      configPackageDBs = [Just (SpecificPackageDB $ sandboxPackageDBPath
                                                       sandboxDir
                                                       compiler
                                                       platform)]
     }
+  where viewSupported = Register.viewSupported compiler conf
+
+
+-- | A general purpose hash function. It is exported as the hash
+-- of sanbox dir needs to be calculated in two different modules.
+-- See http://en.wikipedia.org/wiki/Jenkins_hash_function
```

Comment:
I assume this is the byte-at-a-time variant of that hash function? Can you add a comment explaining the rationale for this choice of hash function?